frontend/src/components/templates/ILab/index.jsx

```diff
@@ -0,0 +1,273 @@
+
+  const {
+    results,
+    tableColumns,
+    activeSortDir,
+    activeSortIndex,
+    page,
+    perPage,
+    summary,
+    tableFilters,
+    filterOptions,
+    categoryFilterValue,
+    filterData,
+    appliedFilters,
+    start_date,
+    end_date,
+    graphData,
+    selectedFilters,
+    totalItems,
+    tableData,
+  } = useSelector((state) => state.ilab);
+  const [expandedResult, setExpandedResult] = useState([]);
+  const [expanded, setAccExpanded] = useState(["bordered-toggle1"]);
+
+  const onToggle = (id) => {
+    const index = expanded.indexOf(id);
+    const newExpanded =
+      index >= 0
+        ? [
+            ...expanded.slice(0, index),
+            ...expanded.slice(index + 1, expanded.length),
+          ]
+        : [...expanded, id];
+    setAccExpanded(newExpanded);
+  };
+  const isResultExpanded = (res) => expandedResult?.includes(res);
+  const setExpanded = async (run, isExpanding = true) => {
+    setExpandedResult((prevExpanded) => {
+      const otherExpandedRunNames = prevExpanded.filter((r) => r !== run.id);
+      return isExpanding
+        ? [...otherExpandedRunNames, run.id]
+        : otherExpandedRunNames;
+    });
+    if (isExpanding) {
+      dispatch(fetchPeriods(run.id));
+      dispatch(fetchMetricsInfo(run.id));
+    }
+  };
+
+  useEffect(() => {
+    if (searchParams.size > 0) {
+      // date filter is set apart
+      const startDate = searchParams.get("start_date");
+      const endDate = searchParams.get("end_date");
+
+      searchParams.delete("start_date");
+      searchParams.delete("end_date");
+      const params = Object.fromEntries(searchParams);
+      const obj = {};
+      for (const key in params) {
+        obj[key] = params[key].split(",");
+      }
+      dispatch(setIlabDateFilter(startDate, endDate, navigate));
+    }
+  }, []);
+
+  useEffect(() => {
+    dispatch(fetchIlabJobs());
+  }, [dispatch]);
+
+  const columnNames = {
+    benchmark: "Benchmark",
+    email: "Email",
+    name: "Name",
+    source: "Source",
+    metric: "Metric",
+    begin_date: "Start Date",
+    end_date: "End Date",
+    status: "Status",
+  };
+
+  //Filter Helper
+  const modifiedTableFilters = useMemo(
+    () =>
+      tableFilters.filter(
+        (item) => item.value !== "endDate" && item.value !== "startDate"
+      ),
+    [tableFilters]
+  );
+  const updateSelectedFilter = (category, value, isFromMetrics = false) => {};
+
+  return (
+    <>
+      <TableFilter
+        tableFilters={modifiedTableFilters}
+        filterOptions={filterOptions}
+        categoryFilterValue={categoryFilterValue}
+        appliedFilters={appliedFilters}
+        filterData={filterData}
+        start_date={start_date}
+        end_date={end_date}
+        selectedFilters={selectedFilters}
+        updateSelectedFilter={updateSelectedFilter}
+        navigation={navigate}
+        type={"ilab"}
+        showColumnMenu={false}
+      />
+      <Table aria-label="Misc table" isStriped>
```

Comment:
It might be helpful to break the table component and its expanded row into smaller, focused components for better readability and maintainability.